python/pyspark/ml/param/__init__.py

```diff
@@ -375,6 +375,18 @@ def copy(self, extra=None):
         that._defaultParamMap = {}
         return self._copyValues(that, extra)
 
+    def set(self, param, value):
+        """
+            Sets the value for a parameter in the parameter map.
+        """
+        if not self.hasParam(param.name):
+            raise TypeError('Invalid param %s given for instance %r. %s' % (p.name, self, e))
```

Comment:
Can you add a test which tries to set this with an invalid Param?  This line will not work since e is not defined, but tests pass since this line of code is not tested.